src/vdf_io/import_vdf/lancedb_import.py

```diff
@@ -0,0 +1,173 @@
+                self.set_dims(namespace_meta, index_name)
+                data_path = namespace_meta["data_path"]
+                final_data_path = self.get_final_data_path(data_path)
+                # Load the data from the parquet files
+                parquet_files = self.get_parquet_files(final_data_path)
+
+                vectors_all = {}
+                for vec_col in namespace_meta.get("vector_columns", []):
+                    vectors_all[vec_col] = {}
+
+                new_index_name = index_name + (
+                    f'_{namespace_meta["namespace"]}'
+                    if namespace_meta["namespace"]
+                    else ""
+                )
+                new_index_name = self.create_new_name(new_index_name, tables)
+                if new_index_name not in tables:
+                    table = self.db.create_table(
+                        new_index_name, schema=pq.read_schema(parquet_files[0])
+                    )
+                else:
+                    table = self.db.open_table(new_index_name)
+
+                for file in tqdm(parquet_files, desc="Iterating parquet files"):
+                    file_path = self.get_file_path(final_data_path, file)
+                    df = self.read_parquet_progress(
+                        file_path,
+                        max_num_rows=(
+                            (self.args.get("max_num_rows") or INT_MAX)
+                            - self.total_imported_count
+                        ),
+                    )
+                    df = cleanup_df(df)
+                    # if there are additional columns in the parquet file, add them to the table
+                    for col in df.columns:
+                        if col not in [field.name for field in table.schema]:
+                            col_type = df[col].dtype
+                            table.add_columns(
+                                {
+                                    col: get_default_value(col_type),
+                                }
+                            )
+                    # split in batches
+                    BATCH_SIZE = self.args.get("batch_size") or 1000
+                    for batch in tqdm(
+                        divide_into_batches(df, BATCH_SIZE),
+                        desc="Importing batches",
+                        total=len(df) // BATCH_SIZE,
+                    ):
+                        if self.total_imported_count + len(batch) >= (
+                            self.args.get("max_num_rows") or INT_MAX
+                        ):
+                            batch = batch[
+                                : (self.args.get("max_num_rows") or INT_MAX)
+                                - self.total_imported_count
+                            ]
+                            max_hit = True
+                        # convert df into list of dicts
+                        batch_dict = batch.to_dict(orient="records")
+                        table.add(batch_dict)
```

Comment:
@AyushExel do you know if this can be made more efficient by directly upserting from the pandas df?